### PR TITLE
[chore] [receiver/apachespark] Use NewDefaultClientConfig instead of manually creating struct

### PR DESCRIPTION
**Pull request by mackjmr**
**Description:**
This PR makes usage of `NewDefaultClientConfig` instead of manually creating the confighttp.ClientConfig struct.

**Link to tracking Issue:** #35457

### CODE REVIEW

File: receiver/apachesparkreceiver/factory_test.go [hunk 1]
```diff
@@ -41,7 +48,12 @@ func TestNewFactory(t *testing.T) {
 						InitialDelay:       time.Second,
 					},
 					ClientConfig: confighttp.ClientConfig{
```
Comment (crobert-1) [inline receiver/apachesparkreceiver/factory_test.go:50]: Shouldn't this follow the same form as other changes, a call to `confighttp.NewDefaultClientConfig()`, then set values that we have defaults for, then use that created default config here?

Comment (mackjmr) [reply receiver/apachesparkreceiver/factory_test.go:50]: I've only been changing to `confighttp.NewDefaultClientConfig()` in non test files, IMO its clearer in tests to have the expected config as manually built struct.

Comment (crobert-1) [reply receiver/apachesparkreceiver/factory_test.go:50]: Doesn't that leave open the possibility of invalid tests (the functional code has a different config than tests), or tests that will just need updated every time default values change?

Comment (mackjmr) [reply receiver/apachesparkreceiver/factory_test.go:50]: > Doesn't that leave open the possibility of invalid tests (the functional code has a different config than tests)

The tests validate that the ClientConfig that is created in the functional code is equal to the ClientConfig that we generate manually in the tests.

> tests that will just need updated every time default values change
If we add a field to the struct and its default is the zero value, then no. If we change a default, then yes we would need to update the value in the manually created struct in tests.

Comment (mackjmr) [reply receiver/apachesparkreceiver/factory_test.go]: I've updated the tests 👍

**Review by djaglowski — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

**Comment by mackjmr** [comment]
@djaglowski @crobert-1 are we good to merge this ?

**Pull request merged**